addon/m3-tracked-array.js

```diff
@@ -25,20 +25,35 @@ export default class M3TrackedArray extends ArrayProxy {
   }
 
   replaceContent(idx, removeAmt, newItems) {
-    newItems = newItems.map((item, idx) => {
+    // Update childModelDatas array
+    // mapping to array of nested models
+    const childModelDatas = this._model._internalModel._modelData.__childModelDatas;
+    const key = this._key;
+    if (childModelDatas && childModelDatas[key] && Array.isArray(childModelDatas[key])) {
```

Comment:
This array check should probably be an assertion

Reply:
Added an assertion.

Reply:
Added an assertion.